examples/assets/scripts/utils/cubemap-renderer.js

```diff
@@ -0,0 +1,108 @@
+// How to use in the Editor:
+// - create entity with Camera component - position of the entity defines where the cubemap is rendered from
+//   and properties of the Camera are used to render cubemap (adjust near / far distance, clearing, layers and other properties)
+//   Note: the layers should contain all layers visible by cybemap camera.
+// - to use generated cube map, you can access it like this using script:
+//   material.cubeMap = entity.script.cubemapRenderer.cubeMap;
+
+var CubemapRenderer = pc.createScript('cubemapRenderer');
+
+CubemapRenderer.attributes.add('resolution', {
+    title: 'Resolution',
+    description: 'Resolution of one side of a cubemap. Use power of 2 resolution if you wish to use Mipmaps.',
+    type: 'number',
+    default: 64
+});
+
+CubemapRenderer.attributes.add('mipmaps', {
+    title: 'Mipmaps',
+    description: 'If set to true, mipmaps will be allocated and autogenerated.',
+    type: 'boolean',
+    default: true
+});
+
+CubemapRenderer.attributes.add('depth', {
+    title: 'Depth',
+    description: 'If set to true, depth buffer will be created.',
+    type: 'boolean',
+    default: true
+});
+
+// initialize code called once per entity
+CubemapRenderer.prototype.initialize = function () {
+
+    // this entity needs to have camera component as well
+    var camera = this.entity.camera;
+    if (!camera) {
+        console.error("CubemapRenderer component requires Camera component to be created on the Entity.");
+        return;
+    }
+
+    // disable camera component, as it's used only as a source of properties
+    camera.enabled = false;
+
+    // limit maximum texture size
+    var resolution = Math.min(this.resolution, this.app.graphicsDevice.maxTextureSize);
```

Comment:
```suggestion
    var resolution = Math.min(this.resolution, this.app.graphicsDevice.maxCubeMapSize);
```

Reply:
thanks, I didn't even know about this one!